src/gui/transferlistwidget.cpp

```diff
@@ -1067,7 +1068,6 @@ void TransferListWidget::displayListMenu()
 
         first = false;
 
-        const bool rechecking = torrent->isChecking();
```

Comment:
IMO it would be better if you revert the changes and do:
```c++
const bool rechecking = torrent->isChecking();
if (rechecking)
    needsPause = true;

const bool queued = (BitTorrent::Session::instance()->isQueueingSystemEnabled() && torrent->isQueued());
```

Reply:
`torrent->isChecking()` means that torrent is in checking state according to libtorrent (i.e. it isn't fully checked now) but it doesn't nececerily mean that it is performing checked at this time.

Reply:
>but it doesn't nececerily mean that it is performing checked at this time.

Even if it is not checking, will it hurt to allow user to pause it?

Reply:
Apparently, my comment mostly refers to the original version, when either pause or resume is available depending on `torrent->isChecking()`.

Reply:
So we should display both resume and pause options if torrent is in checking state regardless of it being paused checking or resumed checking?

Reply:
I mean currently both options are visible on top bar. So shouldn’t those be available in context menu?

Reply:
>So we should display both resume and pause options if torrent is in checking state regardless of it being paused checking or resumed checking?

Can you do some experiments: what happens when a torrent in checking state is paused by the user? does it turns to paused state immediately?
And also what about checking state -> resume? I suppose it should still recheck it before resuming it.

Personally, I don't mind having both resume and pause options regardless of the torrent state.

Reply:
> Personally, I don't mind having both resume and pause options regardless of the torrent state.

This will simplify things.

Recently, rechecking a stopped torrent without changing its "stopped" status no longer seems like a good idea to me. I want to change this behavior in v4.5.

In general, all sorts of implicitness mostly complicate things, IMO.

Reply:
If you pause a checking torrent it shows paused state with the checking percentage. If you resume it, it starts rechecking from where it was left off.

Reply:
> If you pause a checking torrent it shows paused state with the checking percentage. If you resume it, it starts rechecking from where it was left off.

:+1:

Reply:
>Recently, rechecking a stopped torrent without changing its "stopped" status no longer seems like a good idea to me. I want to change this behavior in v4.5.

What do you mean by stopped? you mean paused? Shouldn't it go back to initial state automatically after rechecking is done? The full transition will be: paused -> rechecking -> paused.
Or when paused in the middle of rechecking: paused -> rechecking -> paused (rechecking) -> rechecking (resumed) -> paused.

>If you pause a checking torrent it shows paused state with the checking percentage. If you resume it, it starts rechecking from where it was left off.

What if qbt is exited by the user when paused at rechecking? I suppose the recheck progress goes back to 0 on the next qbt start, right? I'm just pondering on edge cases... and so far this seems to be ok (no bad effects).

Reply:
> I suppose the recheck progress goes back to 0 on the next qbt start, right?

Wrong. The checking progress is saved (of course, provided that the "resume data" was successfully saved at shutdown).


> What do you mean by stopped? you mean paused?

I strongly dislike to use legacy pause/resume terminology in case of user provided states and keep it only for temporary states applied by auto management or queueing systems. I have already stated this repeatedly, and as far as I remember, you agreed with me.

I mean the case when user perform `recheck` on stopped (shown as Paused in GUI) torrent. It will temporary start to check files and then stop again but it doesn't change its state during checking process (it is still stopped).

Reply:
File checking is just one of the step of torrent lifetime. The problems stem from the fact that users often want to apply this step separately, without continuing the further process. So we need to somehow represent such temporarily running torrents. One option was to keep their Stopped state. Now I doubt that this is the best choice, because it does not allow you to distinguish really Stopped torrents from Temporarily running ones. In addition, it makes it unobvious that a temporarily running torrent can be either actually stopped or started in the usual way (i.e. it will continue to work after the file checking is completed).

Reply:
>I strongly dislike to use legacy pause/resume terminology in case of user provided states and keep it only for temporary states applied by auto management or queueing systems. I have already stated this repeatedly, and as far as I remember, you agreed with me.

Even if I agreed with it in the past, (until it is documented in the code base) using the current terminology will avoid misunderstandings when discussing ideas. 

>If you pause a checking torrent it shows paused state with the checking percentage. If you resume it, it starts rechecking from where it was left off.

Now rethinking of it, I find this behavior unexpected. IMO I would expect the pause/resume action to only affect the torrent long-term state. And I find it confusing if the actions can affect recheck (a temporary state) to be paused/resumed.
So when a torrent is rechecking and user invoked pause, I would expect the torrent to be really paused/stopped and the recheck is canceled. And if resume is invoked later, then the torrent will just resume (or recheck if required).

>One option was to keep their Stopped state.

Maybe it is not so bad when the UI can show something like `Paused (rechecking xx%)` or `Downloading (rechecking xx%)`.

Reply:
> Now rethinking of it, I find this behavior unexpected.

I'm afraid many users will disagree with you, including me (especially those who deal with large torrents). It is quite natural that some extended action can be suspended, and then continue further. And file checking is no exception. If you want to restart it, you can always click Recheck again.

Reply:
>Paused (rechecking xx%)

Does it really checking (or queued for checking) currently or just do nothing (i.e. paused during checking process)?

Reply:
I think it's in best interest of the user to provide a function to cancel a recheck or move operation.
Cancelling recheck should reset progress to 0% with a recheck-required on resume.
Cancelling a move operation should move back to original location.

Reply:
> I think it's in best interest of the user to provide a function to cancel a recheck or move operation.

I'm sorry, but how would it help in solving the problem under discussion?

Reply:
I mean it should be a separate option in context menu apart from pause/resume.

Reply:
> I think it's in best interest of the user to provide a function to cancel a recheck or move operation

Won't it add an additional headache by creating even more implicit reactions? For example, what should "Cancel checking" do if applied to a normally running torrent? Just reset the current checking progress? But then it is equivalent to the current Recheck action. Or do you want it to implicitly stop the torrent?

As per moving torrents.
It shouldn't be a problem to cancel queued move. But if the torrent is already in the process of moving, then the only way to do this (without changing the libtorrent) is to schedule a reverse move. 

Anyway, it looks offtopic here. If you want to continue about it, it's better to do it in a separate Issue.

Reply:
>I'm afraid many users will disagree with you, including me (especially those who deal with large torrents). It is quite natural that some extended action can be suspended, and then continue further. And file checking is no exception. If you want to restart it, you can always click Recheck again.

OK, forget it.

As for this PR, I would suggest that making both resume and pause buttons available when rechecking.

Reply:
> As for this PR, I would suggest that making both resume and pause buttons available when rechecking.

:+1:
This is the simplest thing that can be done here without going into the underlying problems.